src/rabin.js

```diff
@@ -0,0 +1,186 @@
+    let nRabin = calculateRabinValueFromKeyParts(p,q);
+    return {
+        "p": decimalToHexString(p),
+        "q": decimalToHexString(q),
+        "nRabin": nRabin
+    };
+}
+
+/**
+ * Creates a Rabin signature of hexadecimal data with a given key's values
+ * @param {String} dataHex Hexadecimal data string value
+ * @param {String} p Key 'p' value
+ * @param {String} q Key 'q' value
+ * @param {String} nRabin Key nRabin value
+ * @returns {JSON} {"signature": String, "paddingByteCount": Number} Signature and padding count
+ */
+function createRabinSignature(dataHex, p, q, nRabin) {
+    // Check if data is valid hex
+    if (!checkIfValidHexString(dataHex))
+        throw ("Error: dataHex %s should be a hexadecimal String with or without '0x' at the beginning.", dataHex);
+    // Remove 0x from data if necessary
+    dataHex = dataHex.replace('0x', '');
+    return root(Buffer.from(dataHex, 'hex'), hexStringToBigInt(p), hexStringToBigInt(q), hexStringToBigInt(nRabin));
+}
+
+/**
+ * Verifies a Rabin signature of hexadecimal data with given padding count, signature and key nRabin value
+ * @param {String} dataHex Hexadecimal data string value
+ * @param {Number} paddingByteCount Padding byte count
+ * @param {String} signatureHex Rabin signature hexadecimal string
+ * @param {String} nRabinHex Key nRabin value hexadecimal string
+ * @returns {Boolean} If signature is valid or not
+ */
+function verifyRabinSignature(dataHex, paddingByteCount, signatureHex, nRabinHex) {
+    // Check if data is valid hex
+    if (!checkIfValidHexString(dataHex))
+        throw ("Error: Data %s should be a hexadecimal String with or without '0x' at the beginning.", dataHex);
+    // Remove 0x from data if necessary
+    dataHex = dataHex.replace('0x', '');
+    // Check if signature is valid hex
+    if (!checkIfValidHexString(signatureHex))
+        throw ("Error: Signature %s should be a hexadecimal String with or without '0x' at the beginning.", signatureHex);
+    // Remove 0x from signature if necessary
+    signatureHex = signatureHex.replace('0x', '');
+    // Check if nRabin is valid hex
+    if (!checkIfValidHexString(nRabinHex))
+        throw ("Error: nRabin %s should be a hexadecimal String with or without '0x' at the beginning.", nRabinHex);
+    // Remove 0x from nRabin if necessary
+    nRabinHex = nRabinHex.replace('0x', '');
+    if(typeof paddingByteCount !== 'number')
+        throw ("Error: paddingByteCount should be a number");
+
+    let dataBuffer = Buffer.from(dataHex, 'hex');
+    let paddingBuffer = Buffer.from('00'.repeat(paddingByteCount), 'hex');
+    let paddedDataBuffer = Buffer.concat([dataBuffer, paddingBuffer]);
+    let dataHash = rabinHashBytes(paddedDataBuffer);
+    let nRabinBigInt = hexStringToBigInt(nRabinHex);
+    let hashMod = dataHash % nRabinBigInt;
+    return hashMod === (hexStringToBigInt(signatureHex) ** 2n % nRabinBigInt);
+}
+
+module.exports = {
+    generateRabinKeyFromSeed,
```

Comment:
The package is already on Rabin, so no need to repeat here. Just rename to
-> generatePrivKey()
-> sign()
-> verify()

Also add a util function, say, privKeyToPubKey(), to get a private key's corresponding public key.

Reply:
Aha I see now!